### ISSUE TITLE
Celery Chord Error API

### ISSUE THREAD

**Issue by npilon** [celery/celery#4963]
## Checklist

- [x] I have included the output of ``celery -A proj report`` in the issue.
      (if you are not able to do this, then at least specify the Celery
       version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.

## Steps to reproduce

Have a chord where one of the tasks in the chord header fails for unrecoverable reasons, but the other tasks succeed.

## Expected behavior

The results of the successful tasks are readily, easily, and conveniently available to the chord `on_error` header.

## Actual behavior

As shown in the test case for preserving chord results when invoking the `on_error` handler, https://github.com/celery/celery/blob/master/t/integration/test_canvas.py#L563-L585 - where `res` is the results of a chord where one task in the chord header failed, something along the lines of the following is needed to extract the results of the successful tasks from the chord header:

```
        # Extract the results of the successful tasks from the chord.
        #
        # We could do this inside the error handler, and probably would in a
        #  real system, but for the purposes of the test it's obnoxious to get
        #  data out of the error handler.
        #
        # So for clarity of our test, we instead do it here.

        # Use the error callback's result to find the failed task.
        error_callback_result = AsyncResult(
            res.children[0].children[0].result[0])
        failed_task_id = error_callback_result.result.args[0].split()[3]

        # Use new group_id result metadata to get group ID.
        failed_task_result = AsyncResult(failed_task_id)
        original_group_id = failed_task_result._get_task_meta()['group_id']

        # Use group ID to get preserved group result.
        backend = fail.app.backend
        j_key = backend.get_key_for_group(original_group_id, '.j')
        redis_connection = get_redis_connection()
        chord_results = [backend.decode(t) for t in redis_connection.lrange(j_key, 0, 3)]
```

**Comment by auvipy** [comment]
please send a pr if you can fix it.

**Comment by npilon** [comment]
Absolutely; but fixing it requires pinning down what the API should be and what API changes are desirable or justifiable here. Should I add it as an optional parameter to the error callback? Or should there be some utility function the error callback has to call to get the results?

**Comment by auvipy** [comment]
if this need design decision could you propose a draft proposal on cep repo?

**Comment by npilon** [comment]
Is there a format that you'd like followed? A template I can use?

**Comment by auvipy** [comment]
https://github.com/celery/ceps check drafts folder for template
